engines/sci/graphics/frameout.cpp

```diff
@@ -55,31 +57,37 @@
 
 namespace Sci {
 
-GfxFrameout::GfxFrameout(SegManager *segMan, ResourceManager *resMan, GfxCoordAdjuster *coordAdjuster, GfxScreen *screen, GfxPalette32 *palette, GfxTransitions32 *transitions) :
+GfxFrameout::GfxFrameout(SegManager *segMan, ResourceManager *resMan, GfxCoordAdjuster *coordAdjuster, GfxPalette32 *palette, GfxTransitions32 *transitions, GfxCursor32 *cursor) :
 	_isHiRes(ConfMan.getBool("enable_high_resolution_graphics")),
 	_palette(palette),
+	_cursor(cursor),
 	_resMan(resMan),
-	_screen(screen),
 	_segMan(segMan),
 	_transitions(transitions),
 	_benchmarkingFinished(false),
 	_throttleFrameOut(true),
 	_throttleState(0),
-	// TODO: Stop using _gfxScreen
-	_currentBuffer(screen->getDisplayWidth(), screen->getDisplayHeight(), nullptr),
 	_remapOccurred(false),
 	_frameNowVisible(false),
-	_screenRect(screen->getDisplayWidth(), screen->getDisplayHeight()),
 	_overdrawThreshold(0),
 	_palMorphIsOn(false) {
 
-	_currentBuffer.setPixels(calloc(1, screen->getDisplayWidth() * screen->getDisplayHeight()));
-
-	// QFG4 is the only SCI32 game that doesn't have a high-resolution toggle
+	// QFG4 is the only SCI32 game that doesn't have a high-resolution version
 	if (g_sci->getGameId() == GID_QFG4) {
 		_isHiRes = false;
 	}
 
+	if (g_sci->getGameId() == GID_PHANTASMAGORIA) {
+		_currentBuffer = Buffer(630, 450, nullptr);
+	} else if (_isHiRes) {
+		_currentBuffer = Buffer(640, 480, nullptr);
+	} else {
+		_currentBuffer = Buffer(320, 200, nullptr);
+	}
```

Comment:
Then that's a horrible ConfigManager design oversight. I thought it was already fixed, I think AGI had a similar issue. This definitely should not be the case. When the option is available, it should be true on default. But when it's not, it should return false. Atm it returns true for all sorts of SCI0->SCI1.1 games, which is of course incorrect.

>I have a somewhat higher confidence about the hashes because they’re automatically generated by code that is 
>known to generate correct hashes

The GUI-Options aren't automatically generated and several other information is also not generated, but added manually by hand.

Anyway, the point is still: this "code" does nothing else than a simple bit-switch. It forces hi-res, nothing else.
So in case hi-res mode was part of the detection-data, I have to assume that you would blindly trust it - on the other hand you don't trust any code, even when it's doing nothing else but a simple bit-switch.

Which to me makes no sense.

You are basically saying "who knows if Gabriel Knight 1 on Mac is really hi-res", although every information available points to that being true. You can't disprove it, because you don't own that version, but whatever. On the other hand you say "but those tables here, those are totally fine, although parts of them were added manually as well, but that data isn't code, so it's surely absolutely correct."

So I guess if clone2727 would have added a GUI-Option for enforcing hi-res, then that would be fine to you and you would trust it. But he added a tiny bit of code, so that's different.